### PR TITLE
Feature/user can search movie by name

### PR DESCRIPTION
**Pull request by rikardomarenzzi**

### CODE REVIEW

File: backend/src/main/java/de/neuefische/rikardo/dub/api/ApiService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public String buildApiUrlToGetMovieDetailsByName(String movieName) {
+        String name = movieName.toLowerCase();
+        String apiUrl = "https://api.themoviedb.org/3/search/";
+        String type = "movie";
+        String apiKeyQuery = "?api_key=";
+        String nameQuery = "&query=";
+
+        apiUrl = apiUrl + type + apiKeyQuery + apiKey + nameQuery + name;
```
Comment (fabianschmauder) [inline backend/src/main/java/de/neuefische/rikardo/dub/api/ApiService.java:28]: i would prefer a direct return here

File: backend/src/main/java/de/neuefische/rikardo/dub/api/ApiService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class ApiService {
+
+    @Value("${tmdb.api.key:defaultApiKeyPlaceholder}")
+    private String apiKey;
+
+    public MovieSearchList getMovieSearchListByName(String movieName) {
+        RestTemplate restTemplate = new RestTemplate();
```
Comment (fabianschmauder) [inline backend/src/main/java/de/neuefische/rikardo/dub/api/ApiService.java:16]: you can inject the template for better testing

File: backend/src/main/java/de/neuefische/rikardo/dub/model/movie/Movie.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Movie {
+
+    @JsonProperty("id")
+    private String id;
```
Comment (fabianschmauder) [inline backend/src/main/java/de/neuefische/rikardo/dub/model/movie/Movie.java:14]: you do not need the Json Property when name name is the saem

File: backend/src/main/java/de/neuefische/rikardo/dub/model/movie/Movie.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    @JsonProperty("title")
+    private String title;
+
+    @JsonProperty("poster_path")
+    private String poster_path;
+
+    /*@JsonProperty("video")
+    private Boolean video;
+
```
Comment (fabianschmauder) [inline backend/src/main/java/de/neuefische/rikardo/dub/model/movie/Movie.java:24]: i would remove comments

File: backend/src/test/java/de/neuefische/rikardo/dub/api/ApiServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        //GIVEN
+        String movieName = "The Matrix";
+        String apiUrl = "https://api.themoviedb.org/3/search/movie?api_key=" + "apiKey" + "&query=the matrix";
+
+        MovieSearchList movieSearchList = new MovieSearchList(new ArrayList<>(List.of(
+                new Movie("603","The Matrix", "/f89U3ADr1oiB1s9GkdPOEpXUk5H.jpg")
+        )));
+
+        ResponseEntity<MovieSearchList> response = new ResponseEntity(movieSearchList,HttpStatus.OK);
+        when(restTemplate.getForEntity(apiUrl, MovieSearchList.class)).thenReturn(response);
```
Comment (fabianschmauder) [inline backend/src/test/java/de/neuefische/rikardo/dub/api/ApiServiceTest.java:39]: you are not testing your api service here 😉 you just test the mock

Comment (fabianschmauder) [reply backend/src/test/java/de/neuefische/rikardo/dub/api/ApiServiceTest.java:39]: this is not an integration test so you can not call the resttemplate

File: backend/src/test/java/de/neuefische/rikardo/dub/api/ApiServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    @DisplayName("The method should return a valid URL query")
+    void buildApiUrlToSearchMovieDetailsByNameTest() {
+        //GIVEN
+        String movieName = "The Matrix";
+        String apiKey = null;
+        //WHEN
+        String result = apiService.buildApiUrlToGetMovieDetailsByName(movieName);
```
Comment (fabianschmauder) [inline backend/src/test/java/de/neuefische/rikardo/dub/api/ApiServiceTest.java:57]: you just make the method public for testing. I would test the url implicit

**Review by fabianschmauder — COMMENTED** [review]
have a look at your tests. 😉

File: backend/src/main/java/de/neuefische/rikardo/dub/api/ApiService.java (hunk 1 above)
Comment (rikardomarenzzi) [reply backend/src/main/java/de/neuefische/rikardo/dub/api/ApiService.java:28]: check

File: backend/src/test/java/de/neuefische/rikardo/dub/api/ApiServiceTest.java (hunk 6 above)
Comment (rikardomarenzzi) [reply backend/src/test/java/de/neuefische/rikardo/dub/api/ApiServiceTest.java:57]: will consider your comments in the next push

**Comment by rikardomarenzzi** [comment]
will consider your comments in the next push

**Pull request merged**